### ISSUE TITLE
A32AA019 is this familiar or similar?

### ISSUE THREAD

**Issue by saikumaru** [RoboDurden/Hoverboard-Firmware-Hack-Gen2.x#62]
Hello!

I would like to use this board through serial control. But haven't been able to identify the IC to start experimenting.

This is what the big chip name reads
XT3011PT
A32AA019

Would appreciate any direction. Thank you!!

Pictures of the board attached 
![IMG_20240216_102145~2](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/44021002/0770a288-8ca8-42e3-b314-a13a38aa9e66)
![IMG_20240216_102116](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/44021002/89174fde-20c3-47d1-affb-ebd0ae0882e7)
![IMG_20240216_101134](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/44021002/df88b7a0-e270-4140-9137-c78dcf13e163)

**Comment by AILIFE4798** [comment]
before posting an issue you should remove all the connectors and rf module blocking the pcb
and use the master board not the slave
but no there seems to be no information about this chip
it could been rebranded chip but the pin number is also wrong
they have really cutten corners on this board theres not much on it
if you are desprate ypu can try to read the chip id anyways with swd programmer and search for that if its an rebranded chip you will be able to find it
if it do not have a programming headder its an asic and not reprogramable
now the mm32spin05 board is supported you should just buy one of those

**Comment by saikumaru** [comment]
Sorry here is a cleaner picture.
![IMG_20240216_140542](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/44021002/38e5e06b-5f6e-474c-a2f1-0a46fed9d4f3)

Am assuming it's the master board because it has a RF receiver on it? Or else the other board too has the same IC on it

**Comment by RoboDurden** [comment]
I am also highly pesimistic about this mcu.
Could you please link the exact module you purchased so we have photos from the outside look/housing and other people will not buy this hoverboard.

The 2.2 layout has a 32 pin version of the gd32f130. If you are extremely lucky, your mcu is pin compatible:
![grafik](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/12238841/50fbea8f-b6a9-40fa-bd66-eec8839d4f08)
PA14 is SWD CLK and PA13 is SWD_IO. You could test with the multimeter in beeping diode-tester mode to see if these two pins go the the 4 pin female dupont header.

**Comment by saikumaru** [comment]
Thank you so much for your reply guys!

This is the exact piece I purchased.
https://hoverpro.in/product/h6-eco-military-blue-hoverboard-with-bluetooth-and-remote/

If there's a way I can identify hoverboard with GD32F130 on this website it would be great. Or if there are links to the products on Amazon, somewhere on the repo/wiki please direct me there.

**Comment by AILIFE4798** [comment]
the empity 4 pin connector may be swd but i cannot tell without enough information
the master board have a power button and 15v regulator, the rf module is on slave to spread the pins to use cheaper micro controllers
a mm32 board is $2 for a set or gd32c8 board is $1 for one piece
i developed the mm32 firmware and robo developed the gd32 firmware so you can decide which one you want
i will admit the gd 32 board have 2 opamp and may support foc in the future while your current board do not even have comparator in the chip

**Comment by saikumaru** [comment]
> P.P.S. as this 4 pin header is for rf module, it will rather connect to a serial uart and not the SWD header. 

No. I meant there's a seperate unused 4pin header(green circled), nothing soldered. And one black with female headers soldered(red circled) which has the RF for serial.

**Comment by AILIFE4798** [comment]
need back of pcb picture to trace

**Comment by RoboDurden** [comment]
I have added a "BEWARE of these hoverboards" folder :

https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/tree/main/BEWARE%20of%20these%20hoverboards

As we have autodetect at least for gd32 by now, it might be worth some further investigation on how to flash these custom mcu - if possible at all.

But i for sure will have not time for that.

@AILIFE4798 please post the $1 / $2 links again. But i thought the gd32 board was only the slave board ?

**Comment by RoboDurden** [comment]
@saikumaru does your hoverboard have plastic wheels or good old aluminium rims ?

Today i fear you can only get complete hoverboards with old gd32f130 as used and no longer brand new :-/

**Comment by AILIFE4798** [comment]
yeah gd32 is only slave and i have not able to get it working yet and i cannot debug gd32f130 for some reason to find whats wrong
i will post link later if it still exist

**Comment by AILIFE4798** [comment]
MM32 aliexpress: de,aliexpress,com/item/1005005959910034.html
MM32 taobao: item,taobao,com/item.htm?id=743458922569
GD32 taobao: item,taobao,com/item.htm?id=722577319258

**Comment by RoboDurden** [comment]
=
MM32 aliexpress: https://de.aliexpress.com/item/1005005959910034.html (does not ship to Germany)
MM32 taobao: https://item.taobao.com/item.htm?id=743458922569
GD32 taobao: https://item.taobao.com/item.htm?id=722577319258

**Comment by saikumaru** [comment]
Ok here is the other board. This has the power switch and also a Bluetooth device powered through it.
So this must be the master board? I hope it gives a better idea now

![IMG_20240216_150928](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/44021002/8a70f167-d15a-4677-870e-981db8a5c43d)

**Comment by saikumaru** [comment]
> @saikumaru does your hoverboard have plastic wheels or good old aluminium rims ?
> 
> Today i fear you can only get complete hoverboards with old gd32f130 as used and no longer brand new :-/

Mine is plastic wheel with rubber tyres. Like the QC sticker says, this one was manufactured in November '23. So definitely new :( but new IC as well

**Comment by AILIFE4798** [comment]
yes this is the master but theres also not a whole lot on it
didnt even see self hold circuitry
i think its using asic unfortunately
not available on chinese website even 
or just rebranded chip

the layout2.21 mm32 board have manufacture date of 20240113

**Comment by AILIFE4798** [comment]
to try prevent people like us flashing it the manufacture will laser off the original marking and write new one
but if the empty header is swd you can try to read the chip ID to find

**Comment by saikumaru** [comment]
> The 2.2 layout has a 32 pin version of the gd32f130. If you are extremely lucky, your mcu is pin compatible: ![grafik](https://private-user-images.githubusercontent.com/12238841/305358252-50fbea8f-b6a9-40fa-bd66-eec8839d4f08.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDgwNzY3MjgsIm5iZiI6MTcwODA3NjQyOCwicGF0aCI6Ii8xMjIzODg0MS8zMDUzNTgyNTItNTBmYmVhOGYtYjZhOS00MGZhLWJkNjYtZWVjODgzOWQ0ZjA4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjE2VDA5NDAyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZlZmYyZmFjNjZiNWQyZmE4MjJkYzIxODE5MzBhYzA1Y2E2Y2Q3NjRkMGM3Y2M4MmEwZmJmM2QxMzBiMGY3MWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.dt2N_JDz1CIC5DeoWUQ2_C4kcZIYUVayQ0qn-9yUZsg) PA14 is SWD CLK and PA13 is SWD_IO. You could test with the multimeter in beeping diode-tester mode to see if these two pins go the the 4 pin female dupont header.


This does match! I am able to beep the pins 23, 24 as you pointed out to the two pins of the 4pin dupont header. Should I just got ahead and solder a SWD connector and follow some suggested steps?

**Comment by AILIFE4798** [comment]
you should read chip ID with pyocd
and the board is 5v so be careful

**Comment by saikumaru** [comment]
> you should read chip ID with pyocd and the board is 5v so be careful

Are these steps documented anywhere please, especially the 5V care? I had played with ESP32/82666/Atmegas in the past, but not something that has an SWD

**Comment by AILIFE4798** [comment]
please read wiki...

**Comment by RoboDurden** [comment]
I think the "Flash via PYocd" wiki page is still missing.

**Comment by AILIFE4798** [comment]
stlink v2 clone can all handle 5v
if you want to use esp32 you need a series resistor 
150R should be enough as the current does not exeed the 20ma rated for protection diode

**Comment by saikumaru** [comment]
> I think the "Flash via PYocd" wiki page is still missing.

Yes, I see the link doesn't exist on wiki. Would you suggest something else please? "Flash via Keil"?

**Comment by AILIFE4798** [comment]
on my stlink it cannot detect target so i assume it wont work

your board actually is like the mm32 board using chip internal comparator for phase current i can see the circuit in your better picture
maybe you can replace it with a mm32 because mm32spin06 is abailable in 32pin
![image](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/142502122/67f8b221-7826-4cf4-9beb-0b89ff34dd60)
unfortunately they not only erased the number, but also the dimple in the chip so i cannot tell which way is up and if the pinout is correct

**Comment by AILIFE4798** [comment]
check the ground and VDD connection, the gd32f130 32pin do not have ground???

**Comment by RoboDurden** [comment]
I think the 32 pin MCU does not have the 32 pins to the side but underneath. Gnd are 9 pads underneath in the middle of the backside.

**Comment by AILIFE4798** [comment]
it is going to be very hard to solder if its the case

**Comment by RoboDurden** [comment]
I think replacing the MCU is not worth the trouble anyway. But if doing it the a little heat gun from the backside is the easiest way anyway I guess. So it does not really makes.a difference if the pins are underneath.

But here the pins are sideways, so maybe your Mm32 32 pin version fits.

Or maybe India has stepped up and can now also produce such simple MCU :-)

**Comment by AILIFE4798** [comment]
yes true the mcu cost more then $1 probably so just get a new board to replace

**Comment by saikumaru** [comment]
I see the pin 32 is connected to ground of 78L05, and also is one of the pins of dupont header. But what should the 4th pin be please?

 I see the 4th pin on header is connected to a sensor on the back, after a resistor. But can't trace it back to MCU anywhere. Should this be a 5V connection?

![17080836273541908631215375418939](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/44021002/a5232dee-0c27-4499-8cbe-66be82d74a47)

**Comment by saikumaru** [comment]
>dimple in the chip so i cannot tell which way is up and if the pinout is correct

The dimple is here, and matched the SWCL and SWIO on pins 24,23 with that as reference.
![17080839860055108665686214361615](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/44021002/40f07939-34f3-4cc0-99b8-ce3027d3b10f)


With name
![17080840223815707883479214334117](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/44021002/908dac89-4e8e-4cd5-b617-3d37d11dd6b7)

**Comment by AILIFE4798** [comment]
the 5v is available on another 7805 pin
pin32 gnd match spin05 32pin package
can you check where is 5v connected to chip
the pin is very thick it should be easy compared to 48pin ones

the swd should not be connected to sensor directly
the remaining one should be 5v
can you post back of board picture like i did at layout2.21 and 2.8 issue
and incase you dont know you can disconnect the motor wire by just pull it out with brute force
so its more convinient to reverse engineer and you can replace with other board

**Comment by saikumaru** [comment]
Here is the back of PCB.

I don't have option to replace with another board, or I dint understand your suggestion, sorry
![Uploading 170808444345097643320245490630.jpg…]()

**Comment by AILIFE4798** [comment]
![image](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/142502122/3e050417-eb66-463d-9b2b-aa5142ad7555)
![image](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/142502122/3898313c-b4e1-4d76-9198-8dda70d89aad)
maybe it really is compatiable

**Comment by AILIFE4798** [comment]
counting from top to bottom
1 and 2 is dio and clk you can try both
3 is gnd
4 is probably 5v, but blocked by sticker on the back of pcb,

**Comment by RoboDurden** [comment]
Would  be nice to see you successfully flash an unkown mcu.
I also found a 48pin mcu on the EcgSolax 60A mppt:
![ecgsolax 60A mppt mcu pins](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/12238841/145b9aab-98c4-4acf-b487-445d5f2df439)
![grafik](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/12238841/addeb407-8df2-49a8-b699-2caa5fd6e3c7)

Might be nice to have an open source 60A mppt hack as well.

**Comment by saikumaru** [comment]
Pin5- 5V(not sure if it's 5, but it comes from a square shape regulator)

Pin 32- ground

Pin 23,24- SWCL, SWIO

So this is a possible clone of mm32spin06?

**Comment by AILIFE4798** [comment]
i have not suceeded with that before but why would you
unless you want to reverse engineer how do they written the foc
because you need to also bypass read protect its not trivial

**Comment by AILIFE4798** [comment]
@RoboDurden you can maybe try this
https://www.eevblog.com/forum/microcontrollers/detect-cpu-via-swd-(segger)/

**Comment by AILIFE4798** [comment]
it is certainly a stm32f103cbtx clone so maybe it is just code compatiable with the original
you should try to run blink on it first

**Comment by RoboDurden** [comment]
@saikumaru , please tell more about your project and what you want to build.
If you are committed to make your new hoverboard never be a hoverboard a gain, then already order some cheap motherboards from the links above for plan B.
And then i guess that simply flashing a blink Keil (free download) example for the different targets GD103 / GD130 / MM32SPIN05 might be the best way to proceed. Maybe you are lucky and you disable read protection by the way and succeed with flashing something that works.

**Comment by AILIFE4798** [comment]
the rom table is different in different mcu 
the debugger do not know how to erase this chip to remove read protect
you can only try different ones


btw @RoboDurden a board can self hold by output to button pin

**Comment by AILIFE4798** [comment]
not all boards can, but my layout2.8.1 can
can your autodetect tell it apart?

**Comment by reed-dan** [comment]
> I think the "Flash via PYocd" wiki page is still missing.

I just posted the [PYocd](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/wiki/Flash-via-PYocd) page. It is based on the MM32SPIN05PF mcu but the concept should apply to most mcus.

**Comment by saikumaru** [comment]
Hi, I have just started using Keil, and this is the error I seem to get with MM32 pack installation.
I know its not a repo specific error, but it would be a great help if you can help me fix it.

![image](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/44021002/84789d71-4611-410e-83bb-a69d4b06abf0)

**Comment by saikumaru** [comment]
This is where I am now. I tried to use the STlink utility to connect but it doesnt respond.
The connections are only the 4 wires from STlink to the board. I havent connected any other ports that I physically unplugged. Also I havent switched on anything on the hoverboard side. Am I doing it correct?


![image](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/44021002/e4af5288-6f63-47fd-8035-7f1ea506fe9c)

**Comment by reed-dan** [comment]
Can you take off the cover of your st link v2 and see what mcu it has? I had really bad luck with my st link v2 until I converted it to jlink. https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/wiki/Converting-St-Link-V2-to-J%E2%80%90LInk

**Comment by RoboDurden** [comment]
Make sure the MCU has the 3.3 or 5 volt it needs.
The pcOcd tools needs the NRST = no reset = inverter reset signal to connect. According to the new wiki page.
You can modify st link dongle to support NRST pin. See readme of this repo here.
I would also install the st link utility and give it a try.

But @AILIFE4798  and @reed-dan  are the experts for connecting and unlocking new boards.

**Comment by RoboDurden** [comment]
The 32 pin only MCU might use the two swd pins for other use like led output on the 2.2 board.
Therefore the NRST pin might be necessary to slow flashing..

**Comment by saikumaru** [comment]
> Can you take off the cover of your st link v2 and see what mcu it has? I had really bad luck with my st link v2 until I converted it to jlink. https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/wiki/Converting-St-Link-V2-to-J%E2%80%90LInk

![1708167784811419499669622860571](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/44021002/af9be7da-d32c-44bf-af38-8217fcdab53a)





> Make sure the MCU has the 3.3 or 5 volt it needs. 

Should I be adding a new 5V supply for just this purpose or can the 5V from USB be sufficient? Just to confirm I haven't connected the battery/switch plugs back. It's just plain SWD headers now.
![17081680393213100222902289393916](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/44021002/c75268fe-08a8-42d9-92cb-2e298dd77019)


>The pcOcd tools needs the NRST = no reset = inverter reset signal to connect. According to the new wiki page. You can modify st link dongle to support NRST pin. See readme of this repo here. I would also install the st link utility and give it a try.


Yes, I have tried with the stlink utility and it didnt work.



Alright, I will try modifying the stlink and see. But just wanted to double check am good till here

**Comment by AILIFE4798** [comment]
get IDCODE error=board bricked :(

**Comment by AILIFE4798** [comment]
because the mcu is only 32pin, they might wanna get all outa it, so swd is disabled
some mcu does not have a bootloader like lks32mc, in that case the mcu is single use and you have to replace it

**Comment by RoboDurden** [comment]
But why then the extra header with the swdclk and swdio pins ?

**Comment by AILIFE4798** [comment]
because you can program it once at the factory when code is not using swd, once they loaded their code that disable swd at startup you cannot flash again

**Comment by saikumaru** [comment]
> get IDCODE error=board bricked :(

Oh no!! Or is it just disabled for reprogramming?
I swapped SWCL SWIO for once as suggested and nothing more odd, hope that dint brick it

**Comment by AILIFE4798** [comment]
it is ok to reverse dio and clk
it will say no target connected when you do so

**Comment by saikumaru** [comment]
Ok what else can I try before I finally give up?

**Comment by RoboDurden** [comment]
No swapping the two swd pins will not brick.
@AILIFE4798 you suggested that the swd is disabled because of only 32 pin. I do not see that as a reason for disabling swd after flashing.

**Comment by AILIFE4798** [comment]
if there is not enough pins the 2 swd pins can be used as normal gpio so swd is disabled

**Comment by RoboDurden** [comment]
And NRST can reactivate it.

**Comment by saikumaru** [comment]
Or could it have been a 3.3V chip? 🤯😯 A and i used 5V from USB

**Comment by AILIFE4798** [comment]
you can measure the voltage when board powered from battery
but there is no 3v3 regulator on board

**Comment by RoboDurden** [comment]
You should have measured the MCU voltage at the MCU pin while using battery and on off button.

**Comment by AILIFE4798** [comment]
> And NRST can reactivate it.

yes,but your code still run and disable it imidiately if you did not put delay
trust me i tried when i bricked my layout2.21

**Comment by saikumaru** [comment]
My bad, yes. I will plug back in and check

For NRST is there a doc I can refer to please?

**Comment by AILIFE4798** [comment]
no docs if you dunno what to look for cuz mystery chip

**Comment by AILIFE4798** [comment]
![image](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/142502122/0bef7818-81e9-4a8b-9140-f5f071f3d8b1)
this is nrst on mm32

**Comment by RoboDurden** [comment]
> yes,but your code still run and disable it imidiately if you did not put delay trust me i tried when i bricked my layout2.21

NRST will prevent the code from starting so swd does not get disabled by user init code.

**Comment by AILIFE4798** [comment]
when nrst is low swd also cannot connect
when it is high the code start running

the way to solve it is using serial method, but not use serial insted still use swd, when in bootloader the swd is always enabled, but we do not know if the mystery chip have a boot0 pin
im pretty sure every micro controller have nrst and is in same plece so it is always worth a try

**Comment by AILIFE4798** [comment]
but when swd header does not include nrst most of the time mean its not needed
if they wrote code that disable swd and reactivate after reset like the layout2.8/2.8.1 they include that pin

**Comment by RoboDurden** [comment]
Maybe the esp32 as flash device the the best way to ensure NRST functionality ?

And the I would try all possible softwares to connect/flash

Did you test the autoflash.bat in the readytoflash/ folder ?
Simply try to upload some binary ..

**Comment by AILIFE4798** [comment]
idk i short the nrst manually and use stlink v2

**Comment by AILIFE4798** [comment]
stlink utilities hate non st or st clone so do not use it
get idcode error mean the wire connected correctly but swd disabled by code
if you use keil it will say internal command error

**Comment by RoboDurden** [comment]
Is there a terminal software that can talk with the swd protocol ?
So that we first see that the MCU is answering at all.
All these flash utilities do a lot in one step and then simply stop with some meaningless error.
Is the swdclk generated by the stlink dongle or the MCU ?

**Comment by AILIFE4798** [comment]
pyocd is that terminal
the gdb server is exactly what you want

**Comment by AILIFE4798** [comment]
swclk is generated by swd probe

**Comment by RoboDurden** [comment]
Well then @saikumaru should test step by step if the MCU is responding at all. Including playing with the NRST pin. And testing every possible pin?

**Comment by AILIFE4798** [comment]
better use oscilscope but half duplex communication always hard to check

**Comment by AILIFE4798** [comment]
you never know who sent it

**Comment by saikumaru** [comment]
> Did you test the autoflash.bat in the readytoflash/ folder ? Simply try to upload some binary ..

No I haven't tested this yet because I couldn't even detect the target chip.

Should I now try ESP as a programmer please, feel free to suggest me next steps

**Comment by AILIFE4798** [comment]
i think you should try to use nrst
either manually with stlink or esp or modify stlink

**Comment by saikumaru** [comment]
I think modifying STlink is better for me. Would you like to suggest me any doc/article for me please ? I googled but dont seem to get if it's just a mid to STlink v2 programmer or any more wiring on board

**Comment by AILIFE4798** [comment]
in bottom of readme

**Comment by AILIFE4798** [comment]
i will copy to wiki some day

**Comment by RoboDurden** [comment]
But @saikumaru has a different stm MCU on his stlink clone.
With an oscilloscope he might be able to detect the pin of the MCU that changes it's state when trying to flash with swd protocol. Then this pin would be needed to connect to the output header of the st link dongle.

**Comment by AILIFE4798** [comment]
first time seeing stm32f101
the genuine ones have stm32f103 but its very hard to get those
the cheapest ones have apm32f103 which is 1:1 clone so pinout is same

**Comment by AILIFE4798** [comment]
your board is manufactured by this company
www.jsxiechang.com/js.html
and yes they laserd off the original marking and printed new one
so not helpful unfortunately

**Comment by AILIFE4798** [comment]
I know the model number of the mcu now I think
I'll tell you tomorrow
It's even more obserd then lks32 so I don't you can get it to work

**Comment by saikumaru** [comment]
> I know the model number of the mcu now I think I'll tell you tomorrow It's even more obserd then lks32 so I don't you can get it to work

Wonderful, that will be very helpful to make decision.

**Comment by saikumaru** [comment]
> first time seeing stm32f101 the genuine ones have stm32f103 but its very hard to get those the cheapest ones have apm32f103 which is 1:1 clone so pinout is same

This one is at least 6-7yrs old, must have been a genuine purchase. Unfortunately I never had chance to play with STM stuff

**Comment by AILIFE4798** [comment]
it may be STC32f12k54

**Comment by AILIFE4798** [comment]
but pinout wise mm32 is closer

**Comment by reed-dan** [comment]
> it may be STC32f12k54

@AILIFE4798 Stc32f12k54 is lqfp48 but this chip is lqfp32
